### PR TITLE
Update Selenium to version 3.141.59 and some clean ups

### PR DESCRIPTION
**Pull request by marcospereira**
I'm unsure which branch I should target here, but my idea is to make this part of the release compatible with 3.1.0. This should be part of updating ScalaTest+Play to use Scalatest 3.1.0.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -23,13 +23,15 @@ developers := List(
...
+scalaVersion := "2.13.1"
 crossScalaVersions := List("2.10.7", "2.11.12", "2.12.10", "2.13.1")
 
 libraryDependencies ++= Seq(
   "org.scalatest" %% "scalatest" % "3.1.0",
-  "org.seleniumhq.selenium" % "selenium-java" % "2.45.0", 
-  "org.eclipse.jetty" % "jetty-server" % "9.4.12.v20180830" % "test",
-  "org.eclipse.jetty" % "jetty-webapp" % "9.4.12.v20180830" % "test"
+  "org.seleniumhq.selenium" % "selenium-java" % "3.141.59",
+  "org.seleniumhq.selenium" % "htmlunit-driver" % "2.36.0",
```
Comment (marcospereira) [inline build.sbt:32]: htmlunit-driver is not a transitive dependency anymore, so it needs to be added explicitly. This is the version compatible with Selenium 3.141.59.

File: src/main/scala/org/scalatestplus/selenium/WebBrowser.scala [hunk 2]
```diff
@@ -1577,7 +1575,7 @@ trait WebBrowser {
         case e: org.openqa.selenium.NoSuchFrameException => 
           throw new TestFailedException(
                      (_: StackDepthException) => Some("Frame at index '" + index + "' not found."),
-                     None,
+                     Option(e),
```
Comment (marcospereira) [inline src/main/scala/org/scalatestplus/selenium/WebBrowser.scala:1578]: I'm not sure the intent here was to ignore the cause, so I'm adding it here and at some places below.

File: src/main/scala/org/scalatestplus/selenium/WebBrowser.scala [hunk 3]
```diff
@@ -4311,8 +4325,7 @@ trait WebBrowser {
         case takesScreenshot: TakesScreenshot => 
           val tmpFile = takesScreenshot.getScreenshotAs(OutputType.FILE)
           val outFile = new File(targetDir, if (fileName.toLowerCase.endsWith(".png")) fileName else fileName + ".png")
-          new FileOutputStream(outFile).getChannel.transferFrom(
```
Comment (marcospereira) [inline src/main/scala/org/scalatestplus/selenium/WebBrowser.scala:4314]: Those were unclosed.

File: src/test/scala/org/scalatestplus/selenium/DriverSpec.scala [hunk 4]
```diff
@@ -34,8 +34,6 @@ class DriverSpec extends funspec.AnyFunSpec {
     }
 
     class GoogleSearchSpecWithChrome extends GoogleSearchSpec with Chrome
-    class GoogleSearchSpecWithSafari extends GoogleSearchSpec with Safari
```
Comment (marcospereira) [inline src/test/scala/org/scalatestplus/selenium/DriverSpec.scala:37]: Unused.

**Comment by cheeseng** [comment]
@marcospereira fyi I am trying this out now.

**Comment by cheeseng** [comment]
@marcospereira 13 of unit tests failed after the changes, perhaps we can create a selenium-3.141 branch to work on this.

**Comment by marcospereira** [comment]
Thanks, @cheeseng. I'm not sure I will have the time to fix all the tests, but if you want, you can use my branch here to add new commits and experiment. :-)

**Comment by marcospereira** [comment]
By the way, my idea about upgrading Selenium version is that [scalatestplus-play is already using 3.141.59](https://github.com/playframework/scalatestplus-play/blob/master/build.sbt#L25) and then we need to align those versions to avoid some dependency hell.

**Comment by marcospereira** [comment]
@cheeseng I was able to track down the problem. I believe it is an issue with HTMLUnit and hence submitted a pull request there:

https://github.com/HtmlUnit/htmlunit/pull/108

If the pull request is accepted there, they will need to cut a new release, then update on Selenium's htmlunit-driver, cut a release as well, and then we can update it here.

**Comment by cheeseng** [comment]
@marcospereira Ah I see, great!  The PR in htmlunit seems to be merged already, will wait for the next release and then we can test and merge this here.

Thank you very much for your help!  Highly appreciated!

**Comment by 987Nabil** [comment]
I am using play 2.7.4 in multiple projects. In one I got an error
```
class org.openqa.selenium.net.UrlChecker tried to access private method com.google.common.util.concurrent.SimpleTimeLimiter.<init>(Ljava/util/concurrent/ExecutorService;)V
```

I found out, that this is caused by the fact that play is using guava 27 and selenium 2.45 guava 25, which gets evicted by sbt.
In 27 the constructor of `SimpleTimeLimiter` that takes an `ExecutorService` is private.
I fixed it by adding a newer selenium explicitly.
```libraryDependencies += "org.seleniumhq.selenium" % "selenium-java" % "3.141.59"```

I document this here, in case other people might have the same problem and hope google will find this post for them :)

@cheeseng maybe this should be mentioned in the scala test docu?

**Comment by cheeseng** [comment]
@marcospereira I tried the latest htmlunit 2.37, the failing tests reduced from 13 tests to 10 tests, wonder if the remaining ones are caused by other issues?

Cheers.

**Comment by timcharper** [comment]
Gentle nudge on this? Lots of dependencies seem to be held up on this. Possible to publish 3.1.0 version without updating selenium?

**Comment by gheine** [comment]
@cheeseng @bvenners any chance this can be released? It's blocking the scalatest 3.1.x upgrade of scalatestplus-play, see https://github.com/playframework/scalatestplus-play/pull/252

**Comment by raboof** [comment]
@gheine I agree it'd be great to be able to upgrade selenium, but the tests suggests upgrading will break getting/settings/clearing some types of fields. I think we'd like to see those problems fixed (or, if it turns out they're problems with the tests, the tests fixed) before upgrading.

**Comment by cheeseng** [comment]
@marcospereira @14mr0n @timcharper @gheine @raboof @bvenners I think this have been quite a while, and probably won't solve with waiting.  I just looked into the failing tests, they are mainly behavior change of range, color, time and datetimelocal input when deal with default value and the value after 'clear' is called.  I have merged this PR into the following branch: 

https://github.com/scalatest/scalatestplus-selenium/tree/selenium-3.141/3.1.1.x

All tests are green now after adjustments to the tests.  I did a sbt publish to sonatype, but I have yet to close and release it as for unknown reason accessing to the Nexus console is unavailable from my side currently.

**Comment by bvenners** [comment]
@marcospereira @14mr0n @timcharper @gheine @raboof @cheeseng I was able to release the artifacts just now. Let us know if you have any trouble with them.

**Comment by nafg** [comment]
What are the maven coordinates?

**Comment by bvenners** [comment]
@nafg, sorry i don't see it on Maven Central yet, it has disappeared from sonatype, and I am unsure if @cheeseng pushed the branch yet. I believe it will be:

group id: org.scalatestplus
arifact id: selenium-3-141
version: 3.1.1.0

Similar to this, but for the higher Selenium version:

https://mvnrepository.com/artifact/org.scalatestplus/selenium-2-45

Once @cheeseng comes back online we'll confirm.

**Comment by bvenners** [comment]
@nafg it is on the README now:

```
libraryDependencies += "org.scalatestplus" %% "selenium-3-141" % "3.1.1.0" % "test"
```

**Pull request closed without merging**